lib/dal/src/cached_module.rs

```diff
@@ -288,271 +349,171 @@ impl CachedModule {
             return Self::latest_user_independent_modules(ctx).await;
         };
 
-        let query = "
-            SELECT DISTINCT ON (schema_id)
-                id,
-                schema_id,
-                schema_name,
-                display_name,
-                category,
-                link,
-                color,
-                description,
-                component_type,
-                latest_hash,
-                created_at,
-                NULL::bytea AS package_data,
-                scoped_to_user_pk
-            FROM cached_modules
-            WHERE scoped_to_user_pk IS NULL OR scoped_to_user_pk = $1
-            ORDER BY schema_id, created_at DESC
-        ";
-
-        let rows = ctx.txns().await?.pg().query(query, &[&user_pk]).await?;
-
-        let mut result = vec![];
-
-        for row in rows {
-            result.push(row.try_into()?);
-        }
+        let query = format!(
+            "
+                SELECT DISTINCT ON (schema_id)
+                    {CACHED_MODULE_LIST_FIELDS}
+                FROM cached_modules
+                WHERE scoped_to_user_pk IS NULL OR scoped_to_user_pk = $1
+                ORDER BY schema_id, created_at DESC
+            "
+        );
 
-        Ok(result)
+        let rows = ctx.txns().await?.pg().query(&query, &[&user_pk]).await?;
+        rows.into_iter().map(TryInto::try_into).try_collect()
     }
 
-    // TO-DO: Paul
-    // I will refactor this + insert together - this is to temporarily get something working
-    async fn insert_private_module(
+    async fn insert(
         ctx: &DalContext,
         module_details: &ModuleDetailsResponse,
         pkg_bytes: Arc<Vec<u8>>,
+        scoped_to_user_pk: Option<UserPk>,
     ) -> CachedModuleResult<Option<Self>> {
-        let bytes_clone = pkg_bytes.clone();
-        let pkg = slow_rt::spawn(async move { SiPkg::load_from_bytes(&bytes_clone) })?.await??;
-
-        let user_pk =
-            UserPk::from_raw_id(Ulid::from_string(module_details.owner_user_id.as_str())?);
-        dbg!(&user_pk);
-        let query = "
-            INSERT INTO cached_modules (
-                schema_id,
-                schema_name,
-                display_name,
-                category,
-                link,
-                color,
-                description,
-                component_type,
-                latest_hash,
-                created_at,
-                package_data,
-                scoped_to_user_pk
-            ) VALUES (
-                $1, $2, $3, $4, $5, $6,
-                $7, $8, $9, $10, $11, $12
-            ) RETURNING
-              id,
-              schema_id,
-              schema_name,
-              display_name,
-              category,
-              link,
-              color,
-              description,
-              component_type,
-              latest_hash,
-              created_at,
-              NULL::bytea AS package_data,
-              NULL::ident AS scoped_to_user_pk
```

Comment:
No longer yielding null here! No one seems to be using the result, but it seems likely you want scoped_to_user_pk to be set to the actual user_pk it was scoped to.